cacheops/redis.py

```diff
@@ -22,12 +25,51 @@ def handle_connection_failure(call):
             warnings.warn("The cacheops cache timed out! Error: %s" % e, RuntimeWarning)
         except redis.RedisError as e:
             logger.exception(e)
+
+    if settings.CACHEOPS_DEGRADE_PERSISTENT_PER_REQUEST:
+        degraded_client_set = set()
+
+
+        class DegradedClientError(redis.RedisError):
+            """ Error which raised after attempt to make a query to the degraded
+            client.
+            """
+
+
+        @decorator
+        def degrade_client_decorator(call):
+            """ Decorator which marks client as degraded if Connection or Timeout
+            error occured during the query.
+            Query to the degraded client won't be executed at all.
+            Degraded mark saves during the request.
+            """
+            if call.self in degraded_client_set:
+                raise DegradedClientError()
+
+            try:
+                return call()
+            except (redis.ConnectionError, redis.TimeoutError):
```

Comment:
I choose only those two errors here, because I can't catch all RedisError.
```cache_thing``` function uses ```client.register_script``` which in its turn uses ```client.evalsha```, but if script does not exists ```evalsha``` raises ```NoScriptError``` (inherited from RedisError as well) that is catched and proccessed in ```register_script```. That's why I cant catch all RedisErrors.
But I surfed through sentry errors and found a lot of ```ReadOnlyError```. Maybe I should catch it as signal for degradation too?
Also decorating only ```get``` and ```evalsha``` methods looks quite strange, although it's enough for usual case of get/set cache. Maybe it's better to decorate ```execute_command``` method?